### PR TITLE
Option for no plot

### LINKED ISSUE #46: Allow deselection of plotting
Allow for not doing any plotting.

Prepare for returning the plot data, not the plot

### PR DESCRIPTION
**Pull request by SRFU-NN**
closes #46

### CODE REVIEW

File: tests/test_optimizer.py [hunk 1]
```diff
@@ -71,6 +71,27 @@ def test_generates_plots_when_run_with_more_than_initialPoints_samples():
...
+    result = optimizer.run(body={
+        "data": sampleData,
+        "optimizerConfig": sampleConfig,
+        "extras": {"graphFormat": "png"}
+    })
+    validateResult(result)
+    assert len(result["result"]["models"]) > 0
+    assert len(result["plots"]) == 2
+
+    # If graphFormat is none, no plots should be returned. This should be faster.
```
Comment (langdal) [inline tests/test_optimizer.py:84]: This test should be split in two - and for good measure, there could be a third test validating that extras={} preserves the current behavior of returning plots

**Review by langdal — APPROVED** [review]
Regarding the issue of putting `graphFormat` into extras. 

This is exactly what the extras field is for. The rule of thumb for extras properties is that they should be optional.

At some point we can do a refactor round and formalise stuff that end up in extras and move them into the checked part of the payload.

File: tests/test_optimizer.py (hunk 1 above)
Comment (SRFU-NN) [reply tests/test_optimizer.py:84]: The third test would be identical to test_generates_plots_when_run_with_more_than_initialPoints_samples().

I am not sure whether it is better to have two identical tests to be explicit about what is tested, or one for speed. Will do both, then you can tell me it is wrong if I shouldn't have.

Comment (SRFU-NN) [reply tests/test_optimizer.py:84]: Done, thanks for the feedback!

**Pull request merged**